moirae/estimators/offline/scipy.py

```diff
@@ -0,0 +1,41 @@
+"""Estimate the state of health using SciPy optimizers"""
+from scipy.optimize import minimize, OptimizeResult
+
+from moirae.estimators.offline.base import OfflineEstimator
+from moirae.estimators.offline.objectives import Objective
+from moirae.models.base import GeneralContainer, HealthVariable
+
+
+class ScipyMinimizer(OfflineEstimator):
+    """Estimate using SciPy's
+    `minimize <https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.minimize.html>`_ function.
+
+    Args:
+        objective: Objective function to be optimized
+        kwargs: Passed to the minimize function
+    """
+
+    def __init__(self,
+                 objective: Objective,
+                 **kwargs):
+        self.objective = objective
+        self.scipy_kwargs = kwargs
+
+    def estimate(self) -> tuple[GeneralContainer, HealthVariable, OptimizeResult]:
+        # Get the scale of the initial error, used to normalize the output
+        x0 = self.objective.get_x0()
+        y0 = self.objective(x0[None, :]).item()  # Should be a scalar
+
+        # Assemble the function call
+        result = minimize(
+            fun=lambda x: self.objective(x[None, :]).item() / max(y0, 1e-12),
```

Comment:
Numerical reasons. I want the values passed to the optimizer to be close to 1, so I make it such that the initial guess is 1. I'm also not 100% on whether all SciPy optimizers use relative or absolute tolerance for their cutoff

Reply:
Seeing your next comment, it seems a bad idea to assume that y is positive. Maybe I need to rethink this or `abs(y0)`

Reply:
Yeah, I figured it was to avoid some numerical difficulties. Would this cause problems if the initial guess is very poor? (for example, such that the computer-precision division by `y0` makes two different cost values appear the same to the optimizer)

Reply:
Great question. I'm not sure. Shall I make it easy to turn "loss scaling" on and off?

Reply:
I think that's a great idea!